applications/FluidDynamicsApplication/custom_utilities/acceleration_limitation_utilities.cpp

```diff
@@ -0,0 +1,116 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "geometries/geometry.h"
+#include "geometries/geometry_data.h"
+#include "utilities/openmp_utils.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+#include "acceleration_limitation_utilities.h"
+
+namespace Kratos
+{
+
+    /* Public functions *******************************************************/
+
+    /**
+     * @brief Iteration over all nodes in ModelPart to find and reduce excessive accelerations.
+     * 
+     * @param rModelPart model part to be controled
+     */
+
+    void AccelerationLimitationUtilities::LimitAccelerationAtNodes(ModelPart &rModelPart) {
+        // Sum the reactions in the model part of interest.
+        // Note that the reactions are assumed to be already computed.
+
+        double dt = rModelPart.GetProcessInfo().GetValue(DELTA_TIME);
+        double time = rModelPart.GetProcessInfo().GetValue(TIME);
+
+        // iterating over all elements
+        #pragma omp parallel for schedule(dynamic) shared(dt, time)
+        for (int i = 0; i < static_cast<int>(rModelPart.Elements().size()); ++i){
+            
+            auto it_elem = rModelPart.ElementsBegin() + i;
+            // iterating over all nodes of the current element
+            for (unsigned int nnode = 0; nnode < it_elem->GetGeometry().size(); nnode++){
+
+                // retrieving velocities of current and last time step
+                array_1d<double, 3> v  = it_elem->GetGeometry()[nnode].FastGetSolutionStepValue( Kratos::VELOCITY, 0 );
+                array_1d<double, 3> vn = it_elem->GetGeometry()[nnode].FastGetSolutionStepValue( Kratos::VELOCITY, 1 );
```

Comment:
And this must be a constant reference `const array_1d<double, 3> vn`

Reply:
Done.